src/ast/ast.c

```diff
@@ -72,6 +72,22 @@ static void _AST_LimitResults(AST *ast, const cypher_astnode_t *root_clause,
 	}
 }
 
+static void _extract_params(const cypher_astnode_t *statement) {
+	rax *params = raxNew();
+	uint noptions =  cypher_ast_statement_noptions(statement);
+	for(uint i = 0; i < noptions; i++) {
+		const cypher_astnode_t *option = cypher_ast_statement_get_option(statement, i);
+		uint nparams = cypher_ast_cypher_option_nparams(option);
+		for(uint j = 0; j < nparams; j++) {
+			const cypher_astnode_t *param = cypher_ast_cypher_option_get_param(option, j);
+			const char *paramName = cypher_ast_string_get_value(cypher_ast_cypher_option_param_get_name(param));
+			const cypher_astnode_t *paramValue = cypher_ast_cypher_option_param_get_value(param);
+			raxInsert(params, (unsigned char *) paramName, strlen(paramName), (void *)paramValue, NULL);
```

Comment:
Is it possible for us to convert parameter values to AR_ExpNodes here? I feel like storing actual constants as the annotation rather than AST pointers would ultimately save us work, especially if a parameter is used in multiple times in the query.

Reply:
you are 100% right. changed